seisbench/models/dkpn.py

```diff
@@ -0,0 +1,784 @@
+
+        model._weights_metadata = weights_metadata
+        model._parse_metadata()
+
+        state_dict = torch.load(weight_path)
+        old_weight = state_dict["inc.weight"]
+        state_dict["inc.weight"] = torch.zeros(
+            old_weight.shape[0], old_weight.shape[1] + 1, old_weight.shape[2]
+        ).type_as(old_weight)
+        state_dict["inc.weight"][:, :3, ...] = old_weight
+        state_dict["inc.weight"][:, 3, ...] = old_weight[:, 0, ...]
+        model.load_state_dict(state_dict)
+        return model
+
+
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================  PREPROC
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+# ====================================================================
+
+
+class PreProc(object):
+    def __init__(self, **kwargs):
+        self.stream = None  # --> Instantiated everytime in the work() class-method
+        self.__dict__.update(kwargs)
+        self.eps = 1e-10
+
+    def __call__(self, state_dict):
+        """
+        Provides an augmentation for seisbench.generate.generator.GenericGenerator.add_augmentations().
+        """
+
+        labels, metadata_labels = state_dict["y"]
+        waveforms, metadata_waveforms = state_dict["X"]  # metadata is the relevant row-field of CSV
+        waveforms = self.__matrix_cfs__(waveforms)
+
+        # ================================  Emulates WINDOW-LEVEL NORM
+        # --- Remove FP stabilization
+        fstab = int(metadata_waveforms["trace_sampling_rate_hz"]*self.fp_stabilization)
+        waveforms = waveforms[:, fstab:(3001+fstab)]
+        labels = labels[:, fstab:(3001+fstab)]
+
+        # --- NORMALIZE  FP-CF matrix with the matrix STD
+        waveforms[0:3, :] = waveforms[0:3, :] / (
+            np.std(waveforms[0:3, :], axis=1, keepdims=True) + 1e-10
+            )
+
+        # --- NORMALIZE  MODULUS matrix with the its STD
+        waveforms[4, :] = waveforms[4, :] / (
+            np.std(waveforms[4, :], axis=-1, keepdims=True) + 1e-10
+            )
+
+        state_dict["X"] = (waveforms, metadata_waveforms)  # OVERRIDE MATRIX --> OUTPUT
+        state_dict["y"] = (labels, metadata_labels)  # OVERRIDE MATRIX --> OUTPUT
+
+    def __matrix_cfs__(self, MM):
```

Comment:
I think this should be `_matrix_cfs` instead as private functions in python typically start with a single underscore and don't use trailing underscores.